src/context.rs

```diff
@@ -379,18 +380,18 @@ impl<'buf> Secp256k1<SignOnlyPreallocated<'buf>> {
     #[inline]
     pub fn preallocate_signing_size() -> usize { Self::preallocate_size_gen() }
 
-    /// Creates a context from a raw context.
+    /// Creates a context from a raw context that can only be used for signing.
     ///
     /// # Safety
     ///
-    /// This is highly unsafe, due to the number of conditions that aren't checked.
-    /// * `raw_ctx` needs to be a valid Secp256k1 context pointer.
-    /// that was generated by *exactly* the same code/version of the libsecp256k1 used here.
-    /// * The capabilities (All/SignOnly/VerifyOnly) of the context *must* match the flags passed to libsecp256k1
-    /// when generating the context.
-    /// * The user must handle the freeing of the context(using the correct functions) by himself.
-    /// * This list *is not* exhaustive, and any violation may lead to Undefined Behavior.
+    /// This is highly unsafe due to the number of conditions that aren't checked, specifically:
     ///
+    /// * `raw_ctx` needs to be a valid Secp256k1 context pointer that was generated by *exactly*
+    ///   the same code/version of the libsecp256k1 used here.
+    /// * The capabilities (`All`/`SignOnly`/`VerifyOnly`) of the context *must* match the flags
+    ///   passed to libsecp256k1 when generating the context.
```

Comment:
This was written before we started fetching size from context during drop so it mandated exact flags. I think today it should be fine to require that at least the signing flag was supplied.

However to my knowledge `secp256k1` asserts validity so it shouldn't lead to UB anyway? OTOH keeping it UB allows us more flexibility. But then wee should explain it's UB for flexibility and "no, we didn't document this incorrectly".

Reply:
I also think that, to the extent we can get away with it, we should avoid mentioning flags since they are going to go away with the next major release. (Since upstream merged https://github.com/bitcoin-core/secp256k1/pull/1126 there are basically no more flags.)

Reply:
Does this mean that the size of context will stay compile-time-known forever? And having a custom mutable context is only needed for signing due to randomization?

Reply:
Both true, yes :).

Reply:
Cool! So in that case we could drop `AlignedType` and make `ffi::Context` to have the size and alignment same as in C using the same trick I made PR with (just twice). That'd simplify many things.

Reply:
I removed the whole bullet point, commenting here to double check we are happy with that?